financial-connections/src/main/java/com/stripe/android/financialconnections/features/networkinglinkverification/NetworkingLinkVerificationViewModel.kt

```diff
@@ -62,41 +62,72 @@ internal class NetworkingLinkVerificationViewModel @AssistedInject constructor(
+    private val getCachedConsumerSession: GetCachedConsumerSession,
 ) : FinancialConnectionsViewModel<NetworkingLinkVerificationState>(initialState, nativeAuthFlowCoordinator) {
 
     init {
         observeAsyncs()
         viewModelScope.launch {
             setState { copy(payload = Loading()) }
-            runCatching { getOrFetchSync().manifest.also { requireNotNull(it.accountholderCustomerEmailAddress) } }
-                .onSuccess { manifest ->
-                    lookupConsumerAndStartVerification(
-                        email = requireNotNull(manifest.accountholderCustomerEmailAddress),
-                        businessName = manifest.businessName,
-                        verificationType = VerificationType.SMS,
-                        onConsumerNotFound = {
-                            analyticsTracker.track(VerificationError(PANE, ConsumerNotFoundError))
-                            navigationManager.tryNavigateTo(InstitutionPicker(referrer = PANE))
-                        },
-                        onLookupError = { error ->
-                            analyticsTracker.track(VerificationError(PANE, LookupConsumerSession))
-                            setState { copy(payload = Fail(error)) }
-                        },
-                        onStartVerification = { /* no-op */ },
-                        onVerificationStarted = { consumerSession ->
-                            val payload = buildPayload(consumerSession, manifest)
-                            setState { copy(payload = Success(payload)) }
-                        },
-                        onStartVerificationError = { error ->
-                            analyticsTracker.track(
-                                VerificationError(PANE, StartVerificationSessionError)
-                            )
-                            setState { copy(payload = Fail(error)) }
-                        }
-                    )
-                }
-                .onFailure { setState { copy(payload = Fail(it)) } }
+
+            runCatching {
+                buildInitData()
+            }.onSuccess {
+                startVerification(it)
+            }.onFailure {
+                setState { copy(payload = Fail(it)) }
+            }
+        }
+    }
+
+    private suspend fun buildInitData(): InitData {
+        val manifest = getOrFetchSync().manifest
+        val isInstantDebits = isLinkWithStripe()
+
+        val email = if (isInstantDebits) {
+            // This pane can appear in two different places in the flow,
+            // and it might not have the consumer session created when launched from the
+            // Link warmup sheet. Therefore, we need to fall back to the customer email.
+            val consumerSession = getCachedConsumerSession()
+            consumerSession?.emailAddress ?: manifest.accountholderCustomerEmailAddress
+        } else {
+            manifest.accountholderCustomerEmailAddress
```

Comment:
[no action] Since it (logically) looks like ID is adding many conditionals of this fashion, we can brainstorm how to make this better in the future (maybe we can have dedicated viewModels for Link vs Networking panes?)